### ISSUE TITLE
Update version of go to >=1.18?

### ISSUE THREAD

**Issue by joshspicer** [dutchcoders/transfer.sh#505]
For personal use, I currently working on adding support for Azure blob storage as an alternative storage driver (https://github.com/dutchcoders/transfer.sh/compare/main...joshspicer:transfer.sh:main?expand=1).  

I would love to contribute back the Azure storage driver back to this repo. The Azure SDKs require at least go 1.18.  I am curious if there's any justification for pinning this project to go 1.15, which is now unsupported.

**Comment by stefanbenten** [comment]
Usually its good to require the latest + the previous to minor versions (ie 1.19, 1.18, 1.17) and thus setting the minimum to 1.17 would be a good intermediate step imo.
For Azure i think there is a "deprecated" sdk that can be used that builds with 1.16 and upwards?

**Comment by aspacca** [comment]
> Usually its good to require the latest + the previous two minor versions (ie 1.19, 1.18, 1.17)

in order to clarify: my goal is to give users the possibility to build from source on any LTS (or equivalent) major system they use, relying on the default repo of the system package manager (or equivalent).

MacOS: hombrew is almost up to latest release, same macports - no problem here
Windows: I honestly don't have a clue :)
Linux:
- Debian, buster, from backports: 1.15
- Anything that can fallback on EPEL, from EPEL7: 1.17

**Comment by aspacca** [comment]
to shorten the discussion:  @joshspicer you should be able to pin to a non-latest version of the azure sdk and still use azure blob storage

is it ok for you?

**Comment by joshspicer** [comment]
FWIW the official stance from the golang team is to [support two major versions](https://go.dev/doc/devel/release) back from the latest (so 1.17, 1.18, and 1.19 today).

I'll see if I can get my changes working with 1.17 so that things are backwards compatible.  I would prefer to use the new Azure Blob Storage sdk (https://github.com/Azure/azure-sdk-for-go).  Not sure if all the features of this SDK require 1.18, or if I can somehow get away with using the SDK in 1.17

**Comment by aspacca** [comment]
as far as I know github.com/Azure/azure-sdk-for-go@v59.0.0+incompatible works with 1.17

**Comment by aspacca** [comment]
@joshspicer are this still an issue for you?

Did you add support for azure blob storage?

**Comment by stefanbenten** [comment]
@aspacca Debian bullseye is the latest release and its backports support 1.19. I dont think we should support anything older than 1.17 anymore, in the best case i would like even only supporting 1.18+. for the same reasons outlined by Josh above.

**Comment by aspacca** [comment]
@stefanbenten but debian LTS is buster, unitl Aug 2024 or something like that

**Comment by stefanbenten** [comment]
Of course, i understand. That said, i do not know many go projects that do act against the recommendation, especially with regards to security patches, etc. Rclone is one of they few that is fairly restrictive and careful about bumping and even they have 1.18 in requirement now..

**Comment by aspacca** [comment]
@stefanbenten oki, let's support only oldest supported version from go
